mu/interface/dialogs.py

```diff
@@ -241,18 +241,6 @@ def setup(self, mode, device_list):
         widget_layout = QVBoxLayout()
         self.setLayout(widget_layout)
 
-        # Check whether esptool is installed, show error if not
-        if not self.esptool_is_installed():
-            error_msg = _(
-                "The ESP Firmware flasher requires the esptool' "
-                "package to be installed.\n"
-                "Select \"Third Party Packages\", add 'esptool' "
-                "and click 'OK'"
-            )
-            error_label = QLabel(error_msg)
-            widget_layout.addWidget(error_label)
-            return
-
```

Comment:
Having thought about this as part of the discussion in the other issues I think it's probably a good idea to keep this as well. 
Users will now be able to remove the esptool package (accidentally or on purpose), so this provides a great fail mode if it's missing.
In fact, maybe we need to add similar fallbacks for the other modes as well.

Reply:
Would that really be possible? Wouldn't the packages be part of the baseline packages that can't be removed? Otherwise users could also remove PyQt I guess...

Reply:
PyQt is part of the baseline packages defined in setup.py that are used to run Mu and can't be removed, but mode packages defined in mu.wheels get installed in the virtualenv and will appear in the settings, so they could be removed.

We probably need to revisit how we expose/explain all these changes to the user, as I think it will be relatively easy to accidentally delete something like pygamezero and then suddenly the mode is no longer in the modes menu.

Reply:
In principle, anything installed as part of the baseline install (including deps of the top-level packages) shouldn't appear in the list of 3rd-party packages. It *is* possible (as I think @dybber pointed out elsewhere) to *add* them via the packaging window and then remove them -- at which point they will be gone from the venv.

Reply:
Yeah, I don't think these mode packages show up in the "Third party packages"-tab unless you type them in yourself. I hope that's the behavior we're going for. I think all packages installed as part of clean installation of Mu, should never be possible to remove. If a potential user-installed dependency requires a different version of one of these package, I actually think we should just throw and error and say that the requested package is incompatible. But I guess that's another discussion...

Reply:
Anyway, I can add it back in, it's just a couple of lines, and it will not do any harm having them there...

Reply:
Sorry, you are right, I thought all the venv packages would appear in the settings window, but I guess we don't show anything stored in the venv.json file.

In that case, yeah, we can leave this out, if we ever need to bring it back in we can just have a look at the git history 👍

Reply:
Okay, ready to merge I guess!